CMS-Enterprise/mint-app
# [MINT-3215] Suggested mto milestones

# MINT-3215

## Description

- new table for suggested milestones
- Various migrations that
    - determine if a milestone should be suggested based on table
    - determine if a section should be suggested based on model plan id
    - enable trigger on task list sections
    - suggest milestones for all legacy data


## How to test this change

1.   To test the migration of legacy data, comment out migration V199
2. Bring up the app and use the postman collection to update plan participants and providers `PlanParticipantsAndProvidersUpdate And make 6 suggested milestones`
3. Run the code of Migration 197 in the sql explorer of your choice to confirm that 6 suggested milestones are created


Alternatively, if you don't comment out V199, running that query will automatically create 6 suggested milestones. You can verify by
1. Looking in SQL
2. Querying common milestones, and verifying that they are suggested 
```
APP_SUPPORT_CON
VET_PROVIDERS_FOR_PROGRAM_INTEGRITY
MANAGE_PROV_OVERLAP
RECRUIT_PARTICIPANTS
REV_SCORE_APP
COMM_W_PART
```
## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
